src/audio/eq_fir.c

```diff
@@ -569,28 +586,43 @@ static int fir_cmd_set_data(struct comp_dev *dev,
 			return -EBUSY;
 		}
 
-		/* Check and free old config */
-		eq_fir_free_parameters(&cd->config);
-
 		/* Copy new config, find size from header */
-		cfg = (struct sof_eq_fir_config *)cdata->data->data;
-		bs = cfg->size;
-		trace_eq("fir_cmd_set_data(): blob size: %u", bs);
-		if (bs > SOF_EQ_FIR_MAX_SIZE || bs == 0)
+		trace_eq("fir_cmd_set_data(): blob size: %u msg_index %u",
+			 cdata->num_elems + cdata->elems_remaining,
+			 cdata->msg_index);
+		if (cdata->num_elems + cdata->elems_remaining >
+		    SOF_EQ_FIR_MAX_SIZE)
 			return -EINVAL;
 
-		/* Allocate buffer for copy of the blob. */
-		cd->config = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM, bs);
-		if (!cd->config) {
-			trace_eq_error("fir_cmd_set_data() error: "
-				       "buffer allocation failed");
-			return -EINVAL;
+		if (cdata->msg_index == 0) {
```

Comment:
Should we also check elems_remaining here as msg->index will be 0 by default on small messages too ?